<?xml version="1.0"?>

<launch>
    <!-- Robot namespace -->
    <arg name="robot_name" default="my_gen3" />
    <!-- app name -->
    <arg name="app_name" default="clutter_pick" />
    <!-- start delay -->
    <arg name="start_delay_seconds" default="0" />
    <!-- gui flag -->
    <arg name="gui" default="false" />
    <!-- arg object type: 0-cubes, 1-real objects -->
    <arg name="object_type" default="1" />
    <!-- arg pick objects or not -->
    <arg name="pick_objects" default="true" />

    <!-- launch load_config -->
    <include file="$(find kinova_apps)/launch/$(arg app_name)/load_config.launch">
        <arg name="app_name" value="$(arg app_name)" />
    </include>

    <!-- launch gui if gui flag is true -->
    <group if="$(eval gui)">
        <include file="$(find kinova_apps)/launch/gui_kinova.launch">
            <arg name="app_name" value="$(arg app_name)" />
            <arg name="pose_list_names" value="joint_angles,sort_place_poses" />
        </include>
    </group>

    <!-- arg for model path -->
    <arg name="model_path" default="$(find kinova_apps)/models/$(arg app_name)/" />
    <!-- arg for model name -->
    <arg name="model_name" default="best_all_object.pt" />

    <!-- launch node with namespace -->
    <group ns="$(arg app_name)">
        <node pkg="kinova_apps" type="clear_clutter_action_test.py" name="clear_clutter_test"
            output="screen">
            <remap from="input_pointcloud_topic" to="/camera/depth_registered/points" />
            <remap from="input_image_topic" to="/camera/color/image_raw" />
            <remap from="camera_info_topic" to="/camera/color/camera_info" />
            <param name="model_path" value="$(arg model_path)" />
            <param name="model_name" value="$(arg model_name)" />
            <param name="object_type" value="$(arg object_type)" />
            <param name="pick_objects" value="$(arg pick_objects)" />
        </node>
    </group>
</launch>